src/components/Slider.js

```diff
@@ -17,27 +18,45 @@ const Slider = () => {
     slidesPerView: 1.5,
   };
 
+  const favoritesContext = useFavorites();
+
+  const toggleFavoriteStatusHandler = (id) => {
+    const isFavorite = favoritesContext.itemIsFavorite(id); //sending id for the checking favorite
+    const userFavorite = products.filter((product) => product.id === id); //getting that spicific item
+    if (isFavorite) {
+      favoritesContext.removeFavorite(id);
+    } else {
+      favoritesContext.addFavorite(userFavorite);
+    }
+  };
   return (
     <section className="products-container">
       <Swiper {...params}>
-        {products.map((singleProduct, index) => {
-          const { img, title, price, id } = singleProduct;
+        {products.map((singleProduct) => {
+          const { img1, title, price, id } = singleProduct;
           return (
-            <Link to={`/item/${id}`} key={id}>
-              <div className="single-product">
-                <img src={img} alt={title} className="headphone-pic" />
-                <span className="like-icon">
-                  <AiOutlineHeart />
-                </span>
-                <div className="product-description">
-                  <h5>{title}</h5>
-                  <h4>{price}</h4>
-                  <span>
-                    <BsArrowRight />
-                  </span>
-                </div>
+            <div className="single-product" key={id}>
+              <Link to={`/item/${id}`}>
+                <img src={img1} alt={title} className="headphone-pic" />
+              </Link>
+              <span
+                className={
+                  favoritesContext.itemIsFavorite(id)
+                    ? `like-icon active`
+                    : `like-icon `
+                }
+                onClick={() => toggleFavoriteStatusHandler(id)}
```

Comment:
This will create function iterating over each item of array map. it is better to create a function inside component like `handleToggleFavorite` which is a function which is returning another function like this:
 ```
 const toggleFavoriteStatusHandler = (id) =>  () => {
    const isFavorite = favoritesContext.itemIsFavorite(id); //sending id for the checking favorite
    const userFavorite = products.filter((product) => product.id === id); //getting that spicific item
    if (isFavorite) {
      favoritesContext.removeFavorite(id);
    } else {
      favoritesContext.addFavorite(userFavorite);
    }
  };

```

then you can use it like this:
`  onClick={toggleFavoriteStatusHandler(id)}`